tests/hash_to_curve_g1.rs

```diff
@@ -0,0 +1,191 @@
+#[cfg(test)]
+#[cfg(feature = "experimental")]
+mod tests {
+    use bls12_381::{
+        hash_to_curve::{ExpandMsgXmd, HashToCurve},
+        G1Affine, G1Projective,
+    };
+    use hex_literal::hex;
+    use sha2::Sha256;
+
+    struct TestCase {
+        msg: &'static [u8],
+        dst: &'static [u8],
+        expected: &'static [u8],
+    }
+
+    impl TestCase {
+        pub fn check_output(&self, output: &[u8]) {
+            if output != self.expected {
+                panic!(
+                    "Test vector result mismatch.\n\
+                    Message: {:x?}\n\
+                    DST: {:x?}\n\
+                    Expected: {:x?}\n\
+                    Found: {:x?}",
+                    self.msg, self.dst, self.expected, output
+                )
+            }
+        }
+    }
+
+    // From <https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16#appendix-J.9.1>
+    #[test]
+    fn hash_to_curve_works_for_draft16_testvectors_g1_sha256_ro() {
+        let dst = b"QUUX-V01-CS02-with-BLS12381G1_XMD:SHA-256_SSWU_RO_";
+
+        let cases = vec![
+            TestCase {
+                msg: b"",
+                dst,
+                expected: &hex!(
+                    "052926add2207b76ca4fa57a8734416c8dc95e24501772c8142787
+                    00eed6d1e4e8cf62d9c09db0fac349612b759e79a1
+                    08ba738453bfed09cb546dbb0783dbb3a5f1f566ed67bb6be0e8c6
+                    7e2e81a4cc68ee29813bb7994998f3eae0c9c6a265"
```

Comment:
Breaking these across lines makes it a bit more difficult to review whether the test cases have changed.

Reply:
Reverted.

Reply:
I believe the wrapping was to make the test vectors look more like in the RFC appendix, but I agree that shouldn't have been done in this PR in the mix with everything else.